srtp/srtp.c

```diff
@@ -2074,6 +2060,154 @@ srtp_remove_stream(srtp_t session, uint32_t ssrc) {
 }
 
 
+srtp_err_status_t
+srtp_update(srtp_t session, const srtp_policy_t *policy) {
+  srtp_err_status_t stat;
+
+  /* sanity check arguments */
+  if ((session == NULL) || (policy == NULL) || (policy->key == NULL))
+    return srtp_err_status_bad_param;
+
+  while (policy != NULL) {
+    stat = srtp_update_stream(session, policy);
```

Comment:
This may be me not understanding the api but I had thought that a session can have multiple streams and that a policy could either apply to a specific stream based on SSRC or to any out bound / inbound stream.
So in this case the policy list will be iterated and each policy will be passed to srtp_update_stream, where that policy will either match a given ssrc and so update that or match the template policy and then update all streams that are associated with that template.
Why anyone would use a list of policies is beyond me but I assume some one must as that is how srtp_create is implemented…

From: John Foley [mailto:notifications@github.com]
Sent: 09 November 2015 20:15
To: cisco/libsrtp
Cc: Pascal Buhler (pabuhler)
Subject: Re: [libsrtp] Add srtp_update functions (#123)


In srtp/srtp.c<https://github.com/cisco/libsrtp/pull/123#discussion_r44316949>:

> @@ -2074,6 +2060,154 @@ srtp_remove_stream(srtp_t session, uint32_t ssrc) {

>  }

>

>

> +srtp_err_status_t

> +srtp_update(srtp_t session, const srtp_policy_t *policy) {

> +  srtp_err_status_t stat;

> +

> +  /* sanity check arguments */

> +  if ((session == NULL) || (policy == NULL) || (policy->key == NULL))

> +    return srtp_err_status_bad_param;

> +

> +  while (policy != NULL) {

> +    stat = srtp_update_stream(session, policy);

Yes, it'll apply all the policies to the session. But isn't only one policy applicable to the session? So the last policy to be applied would be in effect. What's the point of applying all the other policies to the session?

—
Reply to this email directly or view it on GitHub<https://github.com/cisco/libsrtp/pull/123/files#r44316949>.